compute_endpoint/globus_compute_endpoint/endpoint/endpoint_manager.py

```diff
@@ -212,6 +219,38 @@ def wait_for_children(self):
                     log.warning(
                         f"Command terminated by signal: {-rc} ({pid}){proc_args}"
                     )
+
+                ep_name = "<unknown>"
+                for p, n in list(self._running_eps):
+                    if p == pid:
+                        ep_name = n
+                        self._running_eps.discard((p, n))
+
+                cmd_start_args = self._cached_cmd_start_args.pop(pid, None)
+                if not self._time_to_stop and cmd_start_args is not None:
+                    log.info(
+                        "Using cached arguments to start new "
+                        f"instance of user EP (name: {ep_name})"
+                    )
+
+                    try:
+                        cached_rec, args, kwargs = cmd_start_args
+                        updated_rec = pwd.getpwuid(cached_rec.pw_uid)
+                    except Exception as e:
+                        log.warning(
+                            "Unable to update local user information."
+                            f"  ({e.__class__.__name__}) {e}"
+                        )
+                    else:
+                        try:
+                            self.cmd_start_endpoint(updated_rec, args, kwargs)
+                        except Exception:
+                            log.exception(
+                                f"Unable to execute command: cmd_start_endpoint\n"
+                                f"    args: {args}\n"
+                                f"  kwargs: {kwargs}"
+                            )
```

Comment:
As discussed out-of-band, we should probably wrap this code in a function and only call it when the child process terminates normally to avoid crashloops.